openedx/adg/lms/webinars/tests/test_admin.py

```diff
@@ -170,34 +172,68 @@ def test_send_update_emails_to_registrants_in_webinar_update(webinar_admin_insta
+@pytest.mark.parametrize('is_update, is_already_published, is_set_to_be_published', [
+    (False, False, False),
+    (False, False, True),
+    (False, True, False),
+    (False, False, False),
+    (True, False, True),
+    (True, True, True)
+])
+def test_save_related_send_emails(
+    request, webinar_admin_instance, mocker, is_update, is_already_published, is_set_to_be_published
+):
     """
-    Test that upon creating/updating webinar, emails for invites/updated invites are sent.
+    Test that the webinar updation or creation emails are only sent when a webinar is published.
+    No webinar email is sent for a non-published webinar
     """
     mock_send_webinar_emails = mocker.patch('openedx.adg.lms.webinars.admin.send_webinar_emails')
     mock_form_class = mocker.patch('openedx.adg.lms.webinars.forms.WebinarForm')
-
     mock_get_webinar_invitees_emails = mocker.patch('openedx.adg.lms.webinars.admin.get_webinar_invitees_emails')
-    mock_get_webinar_invitees_emails.return_value = ['test@email.com']
+    mock_get_webinar_invitees_emails.return_value = []
 
     request.method = 'POST'
-    mock_form_class.instance = webinar
-
-    if update:
-        user = UserFactory()
-        WebinarRegistrationFactory(user=user, webinar=webinar)
-        WebinarAdmin.save_related(webinar_admin_instance, request, mock_form_class, [], update)
-        mock_send_webinar_emails.assert_has_calls([
-            call(MandrillClient.WEBINAR_UPDATED, webinar, [user.email]),
-            call(MandrillClient.WEBINAR_CREATED, webinar, ['test@email.com'])
-        ])
+
+    presenter = UserFactory()
+    old_webinar = WebinarFactory(is_published=is_already_published, presenter=presenter)
+    webinar_admin_instance.old_webinar = old_webinar
+
+    new_webinar = WebinarFactory(is_published=is_set_to_be_published, presenter=presenter)
+    mock_form_class.instance = new_webinar
+
+    WebinarAdmin.save_related(webinar_admin_instance, request, mock_form_class, [], is_update)
+
+    if not is_update:
+        if is_set_to_be_published:
+            mock_send_webinar_emails.assert_has_calls([
+                call(MandrillClient.WEBINAR_CREATED, new_webinar, [presenter.email])
+            ])
+
+        if is_already_published or not is_set_to_be_published:
+            mock_send_webinar_emails.assert_not_called()
+    else:
+        if is_set_to_be_published:
+            if not is_already_published:
+                mock_send_webinar_emails.assert_has_calls([
+                    call(MandrillClient.WEBINAR_CREATED, new_webinar, [presenter.email])
+                ])
+        else:
+            mock_send_webinar_emails.assert_not_called()
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize('is_published', (False, True))
+def test_is_published_flag_turns_readonly_after_being_set(request, webinar_admin, mocker, is_published):
+    """
+    Tests that the `is_published` flag once set becomes read-only
```

Comment:
*Tests that the `is_published` flag, once set, becomes read-only

OR

*Tests that the `is_published` flag becomes read-only, once set